xdsl/interactive/app.py

```diff
@@ -0,0 +1,250 @@
+"""
+This app allows you to paste xDSL IR into the Input TextArea, select a pass (or multiple passes) to be applied on the input IR
+and subsequently the IR generated by the application of the selected pass(es) is displayed in the Output TextArea. The selected
+passes are displayed in the top right "Selecred passes/query" box. The "Condense" button filters the pass selection list to
+display passes that change the IR, or require arguments to be executed (and thus "may" change the IR). The "Uncondense" button
+returns the selection list to contain all the passes. The "Clear Passes" Button removes the application of the selected passes.
+The "Copy Query" Button allows the user to copy the selected passes/query that they have so far selected (i.e. copy the top right
+box).
+
+This app is still under construction.
+"""
+
+from collections.abc import Callable
+from io import StringIO
+
+# pyright: ignore[reportMissingTypeStubs, reportGeneralTypeIssues]
+from pyclip import copy as pyclip_copy
+from textual import events, on
+from textual.app import App, ComposeResult
+from textual.containers import Horizontal, ScrollableContainer, Vertical
+from textual.widgets import Button, Label, SelectionList, TextArea
+
+from xdsl.dialects import builtin
+from xdsl.ir import MLContext
+from xdsl.parser import Parser
+from xdsl.passes import ModulePass, PipelinePass
+from xdsl.printer import Printer
+from xdsl.tools.command_line_tool import get_all_dialects, get_all_passes
+
+pyclip_copy: Callable[[str], None] = pyclip_copy
+
+
+def transform_input(
+    input_text: str, passes: list[type[ModulePass]]
+) -> builtin.ModuleOp:
+    # Function that takes the input IR, the list of passes to be applied, and applies the passes to the IR.
+    # Returns the module (after pass is applied).
+    ctx = MLContext(True)
+    for dialect in get_all_dialects():
+        ctx.load_dialect(dialect)
+
+    parser = Parser(ctx, input_text)
+    module = parser.parse_module()
+
+    pipeline = PipelinePass([p() for p in passes])
+    pipeline.apply(ctx, module)
+
+    return module
+
+
+def check_if_pass_changes_IR(
```

Comment:
Would it make sense to use the `PipelinePass` for this and use its callback to perform this check here?